airflow-core/src/airflow/ui/src/pages/Connections/ConnectionForm.tsx

```diff
@@ -96,6 +96,19 @@ const ConnectionForm = ({
     mutateConnection(data);
   };
 
+  // Check if extra fields have changed by comparing with initial connection
+  const isExtraFieldsDirty = (() => {
+    try {
+      const initialParsed = JSON.parse(initialConnection.extra) as Record<string, unknown>;
+      const currentParsed = JSON.parse(extra) as Record<string, unknown>;
+
+      return JSON.stringify(initialParsed) !== JSON.stringify(currentParsed);
+    } catch {
+      // If parsing fails, fall back to string comparison
+      return extra !== initialConnection.extra;
+    }
+  })();
```

Comment:
I don't think parsing and stringifying does anything helpful here.

Either we just do `extra !== initialConnection.extra` or we we iterate through initialParsed with Object.keys to actually compare each key/value pair

Reply:
I think it's because of the front-end json formatting. 'content' is actually the same, but UI will show extra '\n' and indent to have a nice display, those wouldn't match with plain string comparison.